### ISSUE TITLE
Mosca does not start when using MongoDB and New Relic agent

### ISSUE THREAD

**Issue by tedhuang** [mcollina/mosca#627]
This is not really a bug directly related to Mosca but it could help explain if someone's running mosca under a project being monitored with the New Relic agent. 

The NodeJS agent for New Relic instruments api calls with the "mongodb" library and replaces the library api functions with its own wrapper function. 

An example of the problem can be found in "lib/persistence/mongo.js :: line 103": 

`that._subscriptions.ensureIndex.bind(that._subscriptions, "client"),`

expects to be calling "bind" on the underlying mongodb api function with the argument "client" and have the callback function implicitly passed in from with "steed.parallel" library but when the newrelic agent is enabled, the function that it is binding to is the wrapper function that the newrelic agent is using to instrument the mongo library so the callback from "steed.parallel" is not called and the task never finishes. 

The solution is to expand `that._subscriptions.ensureIndex.bind(that._subscriptions, "client"),` to 
> function(cb) {
>    that._subscriptions.ensureIndex("client", cb);
>}
but this happens in a lot of instances throughout the file.

Tested with mosca v2.3.0 and newrelic 1.38.2

**Comment by mcollina** [comment]
> The NodeJS agent for New Relic instruments api calls with the "mongodb" library and replaces the library api functions with its own wrapper function.

I do not think this is Mosca's problem. NewRelic should really be transparent. If you want to assemble a PR, I'll be happy to review it.
